### ISSUE TITLE
Composite Tables

### ISSUE THREAD

**Issue by airwedge1** [microsoft/DacFx#362]
**Describe the solution you'd like**
I'd like to be able to extend the definition of tables into multiple files or projects.  For example In SharedProject have a table FooBar and then in MyProject have a file FooBarCustom where the definition is alter table FooBar add MyCustomColumn.  

**Describe alternatives you've considered**
As an alternative I do not use database references instead I include my project as git sub module and just include / exclude individual files to accomplish similar behavior.  However, if a column is added in the SharedProject to a table it needs to be manually added to my custom table definition.

**Comment by dzsquared** [comment]
Making sure I understand this correctly - you want a single table where in 1 project you define a few columns on it, and in another project you define other columns on it.


**shared project**
```sql
create table dbo.FoorBar (
    Id INT PRIMARY KEY,
    SomeColumn NVARCHAR(100)
)
```

**MyProject**
```sql
create table dbo.FoorBar (
    Id INT PRIMARY KEY,
    MyCustomColumn NVARCHAR(25)
)
```
MyProject has a database reference to the shared project



The end result is the table definition from SharedProject is layered under MyProject and any deployment plans are created from a combination of the 2 tables.
if both have a primary key defined that doesn't match, does that cause a build error?

**Comment by airwedge1** [comment]
That is all correct and yes it would cause a build error with more than one primary key, duplicate column names, etc. I would think that it would be easier to do it like the below but either way would work.


shared project

create table dbo.FoorBar (
    Id INT PRIMARY KEY,
    SomeColumn NVARCHAR(100)
)

MyProject

Alter table dbo.FoorBar
Add MyCustomColumn NVARCHAR(25)



________________________________
